packages/windows/src/win-agent.ts

```diff
@@ -0,0 +1,120 @@
+   * @returns A promise resolving to an array of relevant data strings upon successfully achieving the goal.
+   */
+  public async runAsync({ query }: { query: string }): Promise<string[][]> {
+
+    // Get installed apps
+    const apps = await getStartMenuApps();
+    const appTitles = Array.from(apps.keys());
+    console.log(`Found ${appTitles.length} installed apps: ${appTitles.join("\n")}`);
+
+    const appsPlan = await this.navAIGuide.startTaskPlanner_Agent({
+      prompt: sPrompt_App_Planner,
+      userQuery: query,
+      appsSource: appTitles,
+    });
+
+    console.log(`Generated plan with ${appsPlan.steps?.length ?? 0} steps: ${appsPlan.description}`);
+
+    for (const appStep of appsPlan.steps) {
+      const app = apps.get(appStep.appId);
+      if (app == null) {
+        console.error(`Invalid app ID: ${appStep.appId}`);
+      }
+      console.log(`App: ${appStep.appId}, Goal: ${appStep.appEndGoal}`);
+      await this.runAppStepAsync({ app: apps.get(appStep.appId)!, appEndGoal: appStep.appEndGoal });
+    }
+
+    return [];
+  }
+
+  private async runAppStepAsync(appStep: { app: App; appEndGoal: string }): Promise<string[][]> {
+
+    console.log(`Starting at app: ${appStep.app.title}`);
+    const winHandle = await launchAppAsync(appStep.app);
+
+    await new Promise(resolve => setTimeout(resolve, 6000));
+
+    let previousNavAIGuidePage: WindowsNavAIGuidePage | null = null;
+    let currentNavAIGuidePage = await WindowsNavAIGuidePage.fromUIAutomationAsync({
+      windowHandle: winHandle,
+      location: appStep.app.title
+    });
+
+    // saveBase64ImageToFile(currentNavAIGuidePage.screens[0].base64Value);
+
+    this.windowsActionHandler = new WindowsActionHandler(this.navAIGuide);
+    const actions: NLAction[] = [];
+    while (true) {
+      const nextAction = await this.navAIGuide.predictNextNLAction_Visual_Agent({
+        prompt: sPrompt_Predict_Next_NL_Action,
+        previousPage: previousNavAIGuidePage,
+        currentPage: currentNavAIGuidePage,
+        endGoal: appStep.appEndGoal,
+        // scrollable: isViewScrollable(currentNavAIGuidePage.domContent),
+        scrollable: true,
+        previousActions: actions,
+      });
+
+      if (nextAction.previousActionSuccess != null) {
+        console.log(`The previous action was ${nextAction.previousActionSuccess ? "" : "not " }successful: ${nextAction.previousActionSuccessExplanation}`);
+      }
+
+      console.log(`The end goal has ${nextAction.endGoalMet ? "" : "not " }been met: ${nextAction.endGoalMetExplanation}`);
+      if (nextAction.endGoalMet) {
+        return [];
+      }
+
+      console.log(`Predicted next probable action: ${nextAction.actionDescription}`);
+      actions.push(nextAction);
+
+      await this.windowsActionHandler.performAction(nextAction, currentNavAIGuidePage);
+
+      // Delay for a few seconds to allow the action to take effect
+      await new Promise(resolve => setTimeout(resolve, 4000));
```

Comment:
nit: since it is used number of times through codebase, might be prettier to turn it into nice `await sleep(4000);` instead.